framework/aster-frame/src/vm/frame_allocator.rs

```diff
@@ -91,15 +91,16 @@ pub(crate) fn init(regions: &[MemoryRegion]) {
         if region.typ() == MemoryRegionType::Usable {
             // Make the memory region page-aligned, and skip if it is too small.
             let start = region.base().align_up(PAGE_SIZE) / PAGE_SIZE;
-            let end = (region.base() + region.len()).align_down(PAGE_SIZE) / PAGE_SIZE;
+            let end = (region.base().checked_add(region.len()).unwrap()).align_down(PAGE_SIZE)
+                / PAGE_SIZE;
             if end <= start {
                 continue;
             }
             allocator.add_frame(start, end);
             info!(
                 "Found usable region, start:{:x}, end:{:x}",
                 region.base(),
-                region.base() + region.len()
+                region.base().checked_add(region.len()).unwrap()
```

Comment:
Hello @tatetian. Thank you for your review and suggestion. 

It seems that `start` and `end` indicate the page frame range instead of the range of memory region. And this log shows the real addresses.

Thanks again for your careful review.

Reply:
Ok. Then, why bother `checked_add` again?

Reply:
You are right. `region` is not changed, so there is no need to check again.